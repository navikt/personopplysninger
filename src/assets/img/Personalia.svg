<?xml version="1.0" encoding="UTF-8"?>
<svg width="64px" height="64px" viewBox="0 0 64 64" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
    <!-- Generator: Sketch 55 (78076) - https://sketchapp.com -->
    <title>Group 8@1x</title>
    <desc>Created with Sketch.</desc>
    <defs>
        <rect id="path-1" x="0" y="0" width="64" height="64"></rect>
        <circle id="path-3" cx="32" cy="32" r="32"></circle>
    </defs>
    <g id="Page-1" stroke="none" stroke-width="1" fill="none" fill-rule="evenodd">
        <g id="GDPR-Personopplysning" transform="translate(-595.000000, -568.000000)">
            <g id="Group-8" transform="translate(595.000000, 568.000000)">
                <g id="spotlight/-gjenstand-">
                    <mask id="mask-2" fill="white">
                        <use xlink:href="#path-1"></use>
                    </mask>
                    <g id="Background"></g>
                    <g id="spotlight/-element" mask="url(#mask-2)" fill-rule="evenodd" stroke-width="1">
                        <g id="ðŸŒˆ-60">
                            <mask id="mask-4" fill="white">
                                <use xlink:href="#path-3"></use>
                            </mask>
                            <g id="Mask"></g>
                            <g id="NAV-Brand/farge/-BlÃ¥/-60" mask="url(#mask-4)" fill="#99C2E8">
                                <polygon id="blÃ¥-60" points="0 64 64 64 64 0 0 0"></polygon>
                            </g>
                        </g>
                    </g>
                </g>
                <g id="Group-30" transform="translate(12.800000, 19.200000)">
                    <path d="M37.6,0 L0.8,0 C0.3584,0 0,0.3584 0,0.8 L0,26.4 C0,26.8416 0.3584,27.2 0.8,27.2 L7.2,27.2 C7.6416,27.2 8,26.8416 8,26.4 L8,25.2 C8,24.0976 8.8976,23.2 10,23.2 C11.1024,23.2 12,24.0976 12,25.2 L12,26.4 C12,26.8416 12.3584,27.2 12.8,27.2 L25.6,27.2 C26.0416,27.2 26.4,26.8416 26.4,26.4 L26.4,25.2 C26.4,24.0976 27.2976,23.2 28.4,23.2 C29.5024,23.2 30.4,24.0976 30.4,25.2 L30.4,26.4 C30.4,26.8416 30.7584,27.2 31.2,27.2 L37.6,27.2 C38.0416,27.2 38.4,26.8416 38.4,26.4 L38.4,0.8 C38.4,0.3584 38.0416,0 37.6,0 Z" id="line/id-card-1" fill="#FFFFFF" fill-rule="nonzero"></path>
                    <ellipse id="Oval" fill="#337C9B" cx="9.63529412" cy="8" rx="3.23529412" ry="3.2"></ellipse>
                    <path d="M16.1411765,19.2 C16.1411765,15.6653776 13.2441954,12.8 9.67058824,12.8 C6.09698103,12.8 3.2,15.6653776 3.2,19.2 C3.2,19.2 15.7098039,19.2 16.1411765,19.2 Z" id="Oval" fill="#337C9B"></path>
                    <path d="M19.8545455,17.6 L32.9454545,17.6 C33.30695,17.6 33.6,17.9581722 33.6,18.4 C33.6,18.8418278 33.30695,19.2 32.9454545,19.2 L19.8545455,19.2 C19.49305,19.2 19.2,18.8418278 19.2,18.4 C19.2,17.9581722 19.49305,17.6 19.8545455,17.6 Z" id="Rectangle" fill="#3E3832"></path>
                    <path d="M19.8545455,14.4 L32.9454545,14.4 C33.30695,14.4 33.6,14.7581722 33.6,15.2 C33.6,15.6418278 33.30695,16 32.9454545,16 L19.8545455,16 C19.49305,16 19.2,15.6418278 19.2,15.2 C19.2,14.7581722 19.49305,14.4 19.8545455,14.4 Z" id="Rectangle" fill="#3E3832"></path>
                    <path d="M19.8545455,11.2 L32.9454545,11.2 C33.30695,11.2 33.6,11.5581722 33.6,12 C33.6,12.4418278 33.30695,12.8 32.9454545,12.8 L19.8545455,12.8 C19.49305,12.8 19.2,12.4418278 19.2,12 C19.2,11.5581722 19.49305,11.2 19.8545455,11.2 Z" id="Rectangle" fill="#3E3832"></path>
                    <path d="M19.84,6.4 L24.96,6.4 C25.3134622,6.4 25.6,6.7581722 25.6,7.2 C25.6,7.6418278 25.3134622,8 24.96,8 L19.84,8 C19.4865378,8 19.2,7.6418278 19.2,7.2 C19.2,6.7581722 19.4865378,6.4 19.84,6.4 Z" id="Rectangle" fill="#3E3832"></path>
                </g>
            </g>
        </g>
    </g>
</svg>