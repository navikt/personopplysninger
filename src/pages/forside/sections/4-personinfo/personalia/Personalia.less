.underseksjon {
  &__beskrivelse {
    padding: 1rem 0 0 0;
  }
}
