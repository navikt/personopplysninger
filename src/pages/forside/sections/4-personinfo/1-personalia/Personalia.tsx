import React, { Component } from "react";
import Box from "../../../../../components/box/Box";
import personaliaIkon from "../../../../../assets/img/Personalia.svg";
import { Personalia as PersonaliaType } from "../../../../../types/personalia";
import ListElement from "../../../../../components/listelement/ListElement";
import Kilde from "../../../../../components/kilde/Kilde";
import eksternLenkeIkon from "../../../../../assets/img/Link.svg";

interface Props {
  personalia: PersonaliaType;
}

class Personalia extends Component<Props> {
  render() {
    const {
      personident,
      fornavn,
      etternavn,
      spraak,
      statsborgerskap,
      foedested,
      sivilstand,
      kjoenn
    } = this.props.personalia;

    const personidentHeader =
      personident && personident.type === "DNR"
        ? "personalia.dnr"
        : "personalia.fnr";

    const fornavnHeader =
      fornavn && fornavn.indexOf(" ") === -1
        ? "personalia.first_name"
        : "personalia.first_and_middle_name";

    const formattertPersonident =
      personident && personident.verdi.length === 11
        ? {
            ...personident,
            verdi: personident.verdi.replace(/^(.{6})(.*)$/, "$1 $2")
          }
        : personident;

    return (
      <Box
        id="personalia"
        tittel="personalia.tittel"
        beskrivelse="personalia.beskrivelse"
        icon={personaliaIkon}
      >
        <hr className="box__linje-bred" />
        <ul className="list-column-2">
          <ListElement titleId={fornavnHeader} content={fornavn} />
          <ListElement
            className="capitalize"
            titleId="personalia.surname"
            content={etternavn}
          />
          {formattertPersonident && (
            <ListElement
              titleId={personidentHeader}
              content={formattertPersonident.verdi}
            />
          )}
          <ListElement titleId="personalia.language" content={spraak} />
          <ListElement
            titleId="personalia.citizenship"
            content={statsborgerskap}
          />
          <ListElement titleId="personalia.birthplace" content={foedested} />
          <ListElement titleId="personalia.civil_status" content={sivilstand} />
          <ListElement titleId="personalia.gender" content={kjoenn} />
        </ul>
        <Kilde
          kilde="personalia.source.folkeregisteret"
          lenke="https://www.skatteetaten.no/person/folkeregister/"
          lenkeTekst="personalia.link.folkeregisteret"
          eksternLenke={true}
          ikon={eksternLenkeIkon}
        />
      </Box>
    );
  }
}

export default Personalia;
