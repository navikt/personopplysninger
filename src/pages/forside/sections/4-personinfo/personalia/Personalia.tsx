import React, { Component } from "react";
import Box from "../../../../../components/box/Box";
import personalia from "../../../../../assets/img/personalia.svg";
import Kontonummer from "./subsections/Kontonummer";
import Folkeregisteret from "./subsections/Folkeregisteret";
import Telefonnummer from "./subsections/Telefonnummer";
import DKIF from "./subsections/DKIF";
import { Personalia as PersonaliaType } from "../../../../../types/personalia";

interface Props {
  personalia: PersonaliaType;
}

class Personalia extends Component<Props> {
  render() {
    const { kontonr, utenlandskbank } = this.props.personalia;

    return (
      <Box
        id="personalia"
        tittel="personalia.tittel"
        beskrivelse="personalia.beskrivelse"
        icon={personalia}
      >
        <hr className="box__linje-bred" />
        <Folkeregisteret personalia={this.props.personalia} />
        <Kontonummer kontonummer={kontonr} utenlandskBankkonto={utenlandskbank} />
        <Telefonnummer tlfnr={this.props.personalia.tlfnr} />
        <DKIF />
      </Box>
    );
  }
}

export default Personalia;
