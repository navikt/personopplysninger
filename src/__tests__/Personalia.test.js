import * as React from 'react';
import Personalia from 'js/components/Personalia';
import { IntlProvider } from 'react-intl';
const ReactTestRenderer = require('react-test-renderer');
import wrapIntl from 'js/IntlTestHelper';

test('render Personalia with content', () => {
  const properties = {
    navn: {
      datoFraOgMed: '12.08.2008',
      forkortetNavn: 'Ole',
      fornavn: 'Ole',
      kilde: 'string',
      mellomnavn: '',
      slektsnavn: 'Nordmann',
      slektsnavnUgift: 'Nordmann',
    },
    ident: '08048503978',
    statsborgerskap: {
      datoFraOgMed: '12.08.2008',
      kilde: 'D',
      kode: 'Norsk',
    },
    status: {
      datoFraOgMed: '12.08.2008',
      kilde: 'D',
      kode: '',
    },
    telefon: {
      jobb: '12345678',
      jobbDatoRegistrert: '12.08.2010',
      jobbKilde: 'J',
      mobil: '47474747',
      mobilDatoRegistrert: '21.10.2012',
      mobilKilde: 'M',
      privat: '51515151',
      privatDatoRegistrert: '12.06.2008',
      privatKilde: 'P',
    },
    tiltak: {
      datoFraOgMed: '12.03.2018',
      datoTil: '12.03.2023',
      kilde: 'T',
      type: 'Kun telefonkontakt',
    },
    kjonn: 'Mann',
    spraak: {
      datoFraOgMed: '12.03.2009',
      kilde: 'S',
      kode: 'Norsk',
    },
    sivilstand: {
      datoFraOgMed: '12.08.2017',
      kilde: 'S',
      kode: 'Gift',
    },
    kontonummer: {
      datoFraOgMed: '12.08.2019',
      kilde: 'K',
      kode: '12345678911',
    },
    spesreg: {
      datoFraOgMed: '12.08.2019',
      kilde: 'K',
      kode: '12345678911',
    },
  };

  const component = ReactTestRenderer.create(
    wrapIntl(
      <Personalia
        navn={properties.navn}
        ident={properties.ident}
        statsborgerskap={properties.statsborgerskap}
        status={properties.status}
        telefon={properties.telefon}
        tiltak={properties.tiltak}
        kjonn={properties.kjonn}
        spraak={properties.spraak}
        sivilstand={properties.sivilstand}
        kontonummer={properties.kontonummer}
        spesreg={properties.spesreg}
      />));
  expect(component.toJSON()).toMatchSnapshot();
});
