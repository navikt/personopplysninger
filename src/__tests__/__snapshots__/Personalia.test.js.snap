// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Personalia with content 1`] = `
<div
  className="BoxContainer"
>
  <div
    className="BoxWithHeader"
  >
    <h1>
      Personalia
    </h1>
    <div
      className="icon-box-wrapper"
    >
      <img
        alt=""
        className="box-icon"
        src="woman.png"
      />
      <div
        className="Box"
      >
        <ul
          className="personalia-list"
        >
          <li>
            <span
              className="title"
            >
              <span>
                Fornavn
              </span>
            </span>
            <span
              className="content"
            >
              Ole
            </span>
          </li>
          <li>
            <span
              className="title"
            >
              <span>
                Etternavn
              </span>
            </span>
            <span
              className="content"
            >
              Nordmann
            </span>
          </li>
          <li>
            <span
              className="title"
            >
              <span>
                Fødselsnummer
              </span>
            </span>
            <span
              className="content"
            >
              08048503978
            </span>
          </li>
          <li>
            <span
              className="title"
            >
              <span>
                Utenlandsk ID
              </span>
            </span>
            <span
              className="content"
            >
              x
            </span>
          </li>
          <li>
            <span
              className="title"
            >
              <span>
                Telefon
              </span>
            </span>
            <span
              className="content"
            >
              51515151
            </span>
          </li>
          <li>
            <span
              className="title"
            >
              <span>
                Kontonummer
              </span>
            </span>
            <span
              className="content"
            />
          </li>
          <li>
            <span
              className="title"
            >
              <span>
                e-post
              </span>
            </span>
            <span
              className="content"
            >
              x
            </span>
          </li>
          <li>
            <span
              className="title"
            >
              <span>
                Språk
              </span>
            </span>
            <span
              className="content"
            >
              Norsk
            </span>
          </li>
          <li>
            <span
              className="title"
            >
              <span>
                Statsborgerskap
              </span>
            </span>
            <span
              className="content"
            >
              Norsk
            </span>
          </li>
          <li>
            <span
              className="title"
            >
              <span>
                Personstatus
              </span>
            </span>
            <span
              className="content"
            >
              
            </span>
          </li>
          <li>
            <span
              className="title"
            >
              <span>
                Fødested
              </span>
            </span>
            <span
              className="content"
            >
              x
            </span>
          </li>
          <li>
            <span
              className="title"
            >
              <span>
                Spesialregistrering
              </span>
            </span>
            <span
              className="content"
            >
              12345678911
            </span>
          </li>
          <li>
            <span
              className="title"
            >
              <span>
                Sivilstand
              </span>
            </span>
            <span
              className="content"
            >
              Gift
            </span>
          </li>
          <li>
            <span
              className="title"
            >
              <span>
                Sikkerhetstiltak
              </span>
            </span>
            <span
              className="content"
            >
              Kun telefonkontakt
            </span>
          </li>
          <li>
            <span
              className="title"
            >
              <span>
                Kjønn
              </span>
            </span>
            <span
              className="content"
            >
              Mann
            </span>
          </li>
          <li>
            <span
              className="title"
            >
              <span>
                Tilpasset kommunikasjon
              </span>
            </span>
            <span
              className="content"
            >
              x
            </span>
          </li>
        </ul>
        <div
          className="box-footer"
        >
          <span>
            Kilde: NAV, Skatteetaten, Folkeregisteret, Kontakt- og reservasjonsregisteret
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
